01-JS-DOM-APIs/Excercise3b/Promise.js

```diff
@@ -0,0 +1,42 @@
+$(document).ready(function () {
+    $(".hidden-section").fadeIn(1000); // do not pay attention to this. It is from a previous excercise. 
+ });
+
+
+function myfunc( method, url, sync ) {
+    
+    var promise = new Promise(function(resolve, reject){
+        var xmlhttp = new XMLHttpRequest(); 
+        xmlhttp.open(method, url, sync);
+        xmlhttp.onreadystatechange = function() { 
+           
+            if (xmlhttp.readyState === 4 && xmlhttp.status === 200) {
+                resolve(this.response); 
+            }
+            else {
+                reject(error);
+            }
+            
+        };
+        xmlhttp.onerror = function() {
+        reject(Error("Network Error"));
+        };
```

Comment:
this onerror event is handling the reject method for the promise, so you can modify the else statement I marked on the previous comment and remove this, or remove the else from above and leave this event.

Reply:
I've just made the corrections you suggested me. Check it again to see if it's ok. Thanks!